server/chat/backend/agent/tools/bitbucket/utils.py

```
@@ -0,0 +1,175 @@
+    except Exception as e:
+        logger.error(f"Failed to get Bitbucket client for user {user_id}: {e}", exc_info=True)
+        return None
+
+
+def is_bitbucket_connected(user_id: str) -> bool:
+    """Check if Bitbucket credentials exist for a user."""
+    try:
+        from utils.auth.stateless_auth import get_credentials_from_db
+        creds = get_credentials_from_db(user_id, "bitbucket")
+        return bool(creds and creds.get("access_token"))
+    except Exception:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)